README.md

```diff
@@ -19,10 +19,81 @@ Create a `BanyanDBClient` with host, port and a user-specified group and then es
 ```java
 // use `default` group
 client = new BanyanDBClient("127.0.0.1", 17912, "default");
-// establish a connection
+// to send any request, a connection to the server must be estabilished
 client.connect(channel);
 ```
 
+## Schema Management
+
+### Group
+
+To save and fetch data, group and schema must be defined first,
+
+```java
+GroupRegistry groupRegistry = client.groupRegistry();
+// create a group called "default"
+groupRegistry.create("default");
+// list group
+List<String> groups = groupRegistry.list();
+// check existence
+boolean existence = groupRegistry.exist("default");
+```
+
+### Stream
+
+Then we may define a stream with customized configurations,
+
+```java
+StreamMetadataRegistry streamRegistry = client.streamRegistry();
+// build a stream "sw" with 2 shards and ttl equals to 30 days
+Stream s = new Stream("sw", 2, Duration.ofDays(30));
```

Comment:
Which group this stream belongs to?

Reply:
The group is specified while creating the client,

```java
// use `default` group
client = new BanyanDBClient("127.0.0.1", 17912, "default");
```

So the following schema management operations are automatically bounded to the group "default", except group related APIs.

Reply:
It's confused to CRUD groups based on another group's client. Supposing the user connects to an empty instance without any group, how to set the third flag?  

I prefer to introduce a `setGroup` to `client` to manage and switch the current group.

Reply:
Make sense to me. I will refactor this part.

Reply:
Wait a moment, I can stream is created from client, what is the issue of using client group?

Reply:
I think the key point is how to create a client, should it be group binding or sensitive?
I think it it better to have an adminClient and general CRUD client, the latter one could bind group in the initial stage, to ease CRUD.

Reply:
> Wait a moment, I can see stream is created from client, what is the issue of using client group?

I suppose @hanahmily was referring to the `GroupRegistry`. You can CRUD groups other than that one set for the client.

Reply:
I noticed that. That is why I added another comment about separating CRUD client and Admin client.

Reply:
Server usually will provide different username nd password for admin client for keeping security.

Reply:
Notice, generally, only manipulating group and other access grants are admin privilege. Creating stream or measure belong to nornal client.

Reply:
IIUC, we should align whether group CRUD is a part of the client first. 

I thought there're two use cases banyandb have to support:

### Development/Demo
In this case, OAP should play a full-featured client which contains workload resources and group APIs. OAP would take care of creating a group for ingesting and retrieving data. 

### Production
OAP only CRUDs workload resources. A human-being administrator(DBA) will create a group where the OAP writes.

If we should both of them, we have to unify the client to support all functions, introducing a role-based access control( 
 RBAC) to authorize resources to a user/role. This strategy would not affect this client, and RBAC APIs should not belong to this.

Let me wrap this up, I prefer to keep group APIs in this client.

Reply:
For the group, a.k.a. database instance, in BanyanDB. I prefer the `group` could be created default in BanyanDB, like many other databases in quick start mode.
Then other relative APIs could be BanyanDB-CLI's responsibility, which has lower priority for now.

Reply:
If we make java-client as a thin client, removing group APIs sounds good to me.

@lujiajing1126 Please remove `GroupRegistry` from this, then add a `default` group in the prerun phase of the `metadata` module of banyand once Etcd server is ready to serve.

Reply:
> If we make java-client as a thin client, removing group APIs sounds good to me.
> 
> @lujiajing1126 Please remove `GroupRegistry` from this, then add a `default` group in the prerun phase of the `metadata` module of banyand once Etcd server is ready to serve.

Reasonable. I would remove this later

Reply:
Group related APIs are removed